astropy/table/groups.py

```diff
@@ -280,7 +280,9 @@ def aggregate(self, func):
                 and not hasattr(par_col, "mask")
                 and (hasattr(func, "reduceat") or func is np.sum or func is np.mean)
             ):
-                if func is np.mean:
+                if par_col.size == 0:
```

Comment:
I don't think this is quite right - it definitely needs to be outside the above `if` (for non-standard aggregation functions), but I think there may be a more general bug that we get here with `i0s` not being empty to start with...